src/corelibs/U2Gui/src/PasteController.cpp

```diff
@@ -106,18 +106,19 @@ PasteTask *PasteFactoryImpl::createPasteTask(bool isAddToProject) {
     if (mimeData->hasUrls()) {
         return new PasteUrlsTask(mimeData->urls(), isAddToProject);
     }
-    QString clipboardText;
     try {
-        clipboardText = clipboard->text();
+        QString clipboardText = mimeData->hasFormat(U2Clipboard::UGENE_MIME_TYPE)
+                                    ? QString::fromUtf8(mimeData->data(U2Clipboard::UGENE_MIME_TYPE))
+                                    : clipboard->text();
+        if (clipboardText.isEmpty()) {
+            coreLog.error("UGENE can not recognize current clipboard content as one of the supported formats.");
+            return nullptr;
```

Comment:
I don't like the structure here. The natural way out from `try` is `throw`, not `return`. Also I don't see any objects which are able to throw the exception at all, so the reason of the `try` block here is dubious. I offer you whether throw an exception here and catch it in the corresponding block or get rid of "try-catch" at all.

Reply:
This behavior: ("Data in clipboard is too large") was not changed in this PR at all.
We can create another issue to investigate why it was added here originally as a part of separate bug.

Reply:
OK, [done](https://ugene.dev/tracker/browse/UGENE-7060).